src/components/Marquee/Marquee.tsx

```diff
@@ -0,0 +1,174 @@
+import './Marquee.scss'
+
+import React, {Fragment, useEffect, useRef, useState} from 'react'
+
+interface MarqueeProps {
+  /**
+   * Inline style for the container div
+   * Type: object
+   * Default: {}
+   */
+  style?: React.CSSProperties
+  /**
+   * Class name to style the container div
+   * Type: string
+   * Default: ""
+   */
+  className?: string
+  /**
+   * Whether to play or pause the marquee
+   * Type: boolean
+   * Default: true
+   */
+  play?: boolean
+  /**
+   * Whether to pause the marquee when hovered
+   * Type: boolean
+   * Default: false
+   */
+  pauseOnHover?: boolean
+  /**
+   * Whether to pause the marquee when clicked
+   * Type: boolean
+   * Default: false
+   */
+  pauseOnClick?: boolean
+  /**
+   * The direction the marquee is sliding
+   * Type: "left" or "right"
+   * Default: "left"
+   */
+  direction?: 'left' | 'right'
+  /**
+   * Speed calculated as pixels/second
+   * Type: number
+   * Default: 20
+   */
+  speed?: number
+  /**
+   * Duration to delay the animation after render, in seconds
+   * Type: number
+   * Default: 0
+   */
+  delay?: number
+  /**
+   * Whether to show the gradient or not
+   * Type: boolean
+   * Default: true
+   */
+  gradient?: boolean
+  /**
+   * The rgb color of the gradient as an array of length 3
+   * Type: Array<number> of length 3
+   * Default: [255, 255, 255]
+   */
+  gradientColor?: string
+  /**
+   * The width of the gradient on either side
+   * Type: string
+   * Default: 200
+   */
+  gradientWidth?: number | string
+  /**
+   * The children rendered inside the marquee
+   * Type: ReactNode
+   * Default: null
+   */
+  children?: React.ReactNode
+}
+
+export const Marquee: React.FC<MarqueeProps> = ({
+  style = {},
+  className = '',
+  play = true,
+  pauseOnHover = false,
+  pauseOnClick = false,
+  direction = 'left',
+  speed = 20,
+  delay = 0,
+  gradient = true,
+  gradientColor = [255, 255, 255],
+  gradientWidth = 200,
+  children,
+}) => {
+  /* React Hooks */
+  const [containerWidth, setContainerWidth] = useState(0)
+  const [marqueeWidth, setMarqueeWidth] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const [isMounted, setIsMounted] = useState(false)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const marqueeRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    setIsMounted(true)
+  }, [])
+
+  useEffect(() => {
```

Comment:
este jedna vec, na tomto riadku krici eslint, ze useEffectu chyba dependency array. ked tam dependency array nie je, pusta sa to pri kazdom renderi. kedze je to zda sa pointa toho komponentu, ze sa po kazdom renderi upravia hodnoty (a ked som skusil ponukany array tam pridat, prestalo to scrollovat 😄 ), tak treba aspon pridat:
```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
```